src/main/java/com/flansmod/common/guns/AttachmentType.java

```diff
@@ -124,133 +127,96 @@ public AttachmentType(TypeFile file)
 	}
 	
 	@Override
-	protected void read(String[] split, TypeFile file)
-	{
-		super.read(split, file);
-		try
-		{
-			if(split[0].equals("AttachmentType"))
-				type = EnumAttachmentType.get(split[1]);
-			else if(FMLCommonHandler.instance().getSide().isClient() && (split[0].equals("Model")))
-				model = FlansMod.proxy.loadModel(split[1], shortName, ModelAttachment.class);
-			else if(split[0].equals("ModelScale"))
-				modelScale = Float.parseFloat(split[1]);
-			else if(split[0].equals("Texture"))
-				texture = split[1];
-			
-			else if(split[0].equals("Silencer"))
-				silencer = Boolean.parseBoolean(split[1].toLowerCase());
-			else if(split[0].equals("DisableMuzzleFlash") || split[0].equals("DisableFlash"))
-				disableMuzzleFlash = Boolean.parseBoolean(split[1].toLowerCase());
-			
+	protected void read(ConfigMap config, TypeFile file) {
+		try {
+			super.read(config, file);
+
+			String typeString = ConfigUtils.configString(config, "AttachmentType", type.toString());
+			type = EnumAttachmentType.get(typeString);
+
+			model = FlansMod.proxy.loadModel(modelString, shortName, ModelAttachment.class);
```

Comment:
Same here, referencing the model class server-side might cause an exception and not load the remainder of the config file.